### PR TITLE
harprofiler ami creation via packer.

### PR DESCRIPTION
**Pull request by benpatterson**
TE-583

### CODE REVIEW

**Comment by benpatterson** [comment]
@jzoldak @cgoldberg @clytwynec

File: playbooks/roles/harprofiler/tasks/main.yml [hunk 1]
```diff
@@ -38,3 +38,17 @@
     regexp="browsermob_dir"
     line="browsermob_dir: /usr/local"
     state=present
+
+- name: create validation shell script
+  template:
+    owner: "{{ harprofiler_user }}"
+    src: validate_harprofiler_install.sh.j2
```
Comment (jzoldak) [inline playbooks/roles/harprofiler/tasks/main.yml:45]: where is this template?

File: util/packer/harprofiler.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    "playbook_remote_dir": "/tmp/packer-edx-playbooks",
+    "ami": "{{env `HARPROFILER_BASE_AMI`}}"
+  },
+  "builders": [{
+    "type": "amazon-ebs",
+    "access_key": "{{user `aws_access_key`}}",
+    "secret_key": "{{user `aws_secret_key`}}",
+    "region": "us-east-1",
+    "source_ami": "{{user `ami`}}",
+    "instance_type": "m3.large",
```
Comment (jzoldak) [inline util/packer/harprofiler.json]: I thought that the harprofiler has a pretty small footprint, can we use a micro?

File: playbooks/roles/harprofiler/tasks/main.yml (hunk 1 above)
Comment (benpatterson) [inline playbooks/roles/harprofiler/tasks/main.yml:45]: I hadn't checked it in. Fixed.

File: util/packer/harprofiler.json (hunk 2 above)
Comment (benpatterson) [inline util/packer/harprofiler.json]: Can't use a micro, but we can use an m1.small. FWIW, this is just the size of machine for doing the AMI building itself. Once we have the AMI, we can put it on whatever class of machine we'd want.

File: playbooks/roles/harprofiler/templates/validate_harprofiler_install.sh.j2 [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+
+# This script confirms that harprofiler can successfully run on the
+# target machine.
+source {{ harprofiler_venv_dir }}/bin/activate
+cd {{ harprofiler_dir }}
```
Comment (clytwynec) [inline playbooks/roles/harprofiler/templates/validate_harprofiler_install.sh.j2:7]: Why is this line needed if `chdir={{ harprofiler_dir }}` is used?

Comment (benpatterson) [inline playbooks/roles/harprofiler/templates/validate_harprofiler_install.sh.j2:7]: I left it in there in case someone ends up calling the shell not-through ansible...

File: util/packer/harprofiler.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "type": "file",
+    "source": "../../requirements.txt",
+    "destination": "{{user `playbook_remote_dir`}}/requirements.txt"
+  }, {
+    "type": "shell",
+    "inline": ["cd {{user `playbook_remote_dir`}}",
+      "virtualenv packer-venv",
+      ". packer-venv/bin/activate",
+      "pip install -q -r requirements.txt",
+      "echo '[jenkins_worker]' > inventory.ini",
```
Comment (clytwynec) [inline util/packer/harprofiler.json]: Should this really say 'jenkins_worker' here?

File: playbooks/roles/harprofiler/templates/validate_harprofiler_install.sh.j2 (hunk 3 above)
Comment (clytwynec) [inline playbooks/roles/harprofiler/templates/validate_harprofiler_install.sh.j2:7]: Ok. Makes sense.

File: util/packer/harprofiler.json (hunk 4 above)
Comment (benpatterson) [inline util/packer/harprofiler.json]: Good catch. Fixed it.

**Comment by clytwynec** [comment]
:+1:
